app/controllers/api/mixins/chargeback_assignment.rb

```diff
@@ -104,35 +100,37 @@ def chargeback_rate(parameter_record)
         @chargeback_rate[rate_id] ||= ChargebackRate.find(rate_id)
       end
 
-      def tag_target_assignment(record, _assignment_type, rate_type)
-        target = tag_assigment(record, rate_type)
-        [target, rate_type == "Storage" ? 'storage' : record['assignment_prefix']]
-      end
-
-      def label_target_assignment(record, assignment_type, rate_type)
-        target = target_from(record['href'], assignment_type, rate_type)
-        target_add_label_defaults(target)
+      def tag_target_assignment(parameter_record, rate, rate_type)
+        record = parameter_record["tag"]
+        rate_tag = rate_type == "Storage" ? 'storage' : record['assignment_prefix']
+        {:cb_rate => rate, :tag => [tag_assigment(record, rate_type), rate_tag]}
       end
 
-      def resource_target_assignment(record, assignment_type, rate_type)
-        target_from(record['href'], assignment_type, rate_type)
+      def label_target_assignment(parameter_record, rate, rate_type)
+        record = parameter_record["label"] = parameter_record.delete("resource")
+        {:cb_rate => rate, :label => [target_from(record['href'], "label", rate_type), "container_image"]}
       end
 
-      def target(parameter_record, assignment_type, rate_type)
-        target_assignment_method = "#{assignment_type}_target_assignment"
-        send(target_assignment_method, parameter_record[assignment_type.to_s], assignment_type, rate_type)
+      def resource_target_assignment(parameter_record, rate, rate_type)
+        record = parameter_record["resource"]
+        {:cb_rate => rate, :object => target_from(record['href'], "resource", rate_type)}
       end
 
       def convert_assignment_key_from(parameter_key)
         parameter_key == 'resource' ? :object : parameter_key.to_sym
       end
 
       def rate_assignment(parameter_record, assignment_type, rate_type)
-        parameter_record['label'] = parameter_record.delete('resource') if assignment_type == :label
-
         rate = chargeback_rate(parameter_record)
 
-        {:cb_rate => rate, convert_assignment_key_from(assignment_type) => target(parameter_record, assignment_type, rate_type)}
```

Comment:
was this the only caller of the `target` method?

Reply:
Note, I don't like how the old code refers to a local `target` variable while we also had that method.  It's minor but there's no reason to use the same name from a different scope.  I'm glad that's going away.

Reply:
after these changes, there are no callers for target `ag --ruby '\btarget\b'`